lib/git.js

```diff
@@ -427,14 +428,30 @@ class Git extends EventEmitter {
    * @method listen
    * @memberof Git
    * @param  {Number}   port     - the port to start the server on
+   * @param  {Array}   port     - the https options
    * @param  {Function} callback - the function to call when server is started or error has occured
    */
-  listen(port, callback) {
-      var self = this;
-      this.server = http.createServer(function(req, res) {
+  listen(port, options, callback) {
+    if(typeof options == 'function') {
+      callback = options;
+    }
+
+    var self = this;
+    this.server = http.createServer(function(req, res) {
+        self.handle(req, res);
+    });
+    this.server.listen(port, function() {
+      callback(port, 'http');
+    });
+
+    if(typeof options != 'function') {
```

Comment:
This case can be avoided if there is a default options provided if none is.

Reply:
What Do You Mean By This Because I Wrote It That Way So it Someone Doesn't Want HTTPS They Can Just Ignore It And Use This Module Like They Used To?